web/src/components/QRCodeModal/QRCodeModal.tsx

```diff
@@ -0,0 +1,45 @@
+import './QRCodeModal.css'
+import QRCode from './QRCode';
+import { useEffect } from 'react';
+
+
+interface QRCodeModalProps {
+    showQR: boolean;
+    setShowQR: React.Dispatch<React.SetStateAction<boolean>>;
+    qrURL: string;
+  }
+
+
+export default function QRCodeModal({ showQR, setShowQR, qrURL }: QRCodeModalProps) {
+
+    useEffect(() => {
+        if (showQR) {
+          document.body.classList.add('modal-open');
+        } else {
+          document.body.classList.remove('modal-open');
+        }
+        return () => {
+          document.body.classList.remove('modal-open');
+        };
```

Comment:
This logic seems a bit redundant. When the else branch is reached the `remove` is called twice. Is that behavior intended? If not I would refactor to:

```suggestion
        if (showQR) {
          document.body.classList.add('modal-open');
        }
        return () => {
          document.body.classList.remove('modal-open');
        };
```